File: web-service/security/src/main/java/io/ehdev/conrad/security/jwt/JwtManager.java
package io.ehdev.conrad.security.jwt;

import io.ehdev.conrad.api.user.database.BaseUserModel;
import io.ehdev.conrad.security.database.model.TokenType;

import java.time.LocalDateTime;

public interface JwtManager {

    String createUserToken(BaseUserModel user, LocalDateTime expiration);

    String createToken(BaseUserModel user, TokenType type, LocalDateTime expiration);

    UserToken parseToken(String cookieValue);
}

File: auth/authentication/src/main/kotlin/tech/crom/security/authentication/jwt/JwtManager.kt
package tech.crom.security.authentication.jwt

import tech.crom.database.api.TokenManager
import tech.crom.model.repository.CromRepo
import tech.crom.model.user.CromUser


interface JwtManager {

    fun createUserToken(user: CromUser): String

    fun createRepoToken(repo: CromRepo): String

    fun createToken(tokenDetails: TokenManager.TokenDetails): String

    fun parseToken(token: String?): JwtTokenAuthentication?

}
